.github/workflows/desktop-build.yml

```diff
@@ -239,105 +239,138 @@ jobs:
-    continue-on-error: ${{matrix.allow-failure == 'yes'}}
+    runs-on: ${{ matrix.runner }}
     env:
-      # Common parameters for all macOS builds
-      QT_VERSION: 6.6.*
-      QT_ARCH: clang_64
-      QT_MODULES: "qtimageformats qtmultimedia qtwebsockets"
-      # Build-specific environment variables
-      CCACHE_DIR: ${{github.workspace}}/.ccache/${{matrix.os}}-${{matrix.type}}
-      CCACHE_SIZE: 500M
-      DEVELOPER_DIR:
-        /Applications/Xcode_${{matrix.xcode}}.app/Contents/Developer
-      CMAKE_GENERATOR: 'Ninja'
+      # Common parameters for all builds
+      QT_VERSION: ${{ matrix.qt_version || '6.6.*' }}
+      QT_ARCH: ${{ matrix.os == 'macOS' && 'clang_64' || 'win64_msvc2019_64' }}
+      QT_MODULES: ${{ matrix.qt_version != '5.15.*' && 'qtimageformats qtmultimedia qtwebsockets' || '' }}
+      CACHE_QT: ${{ matrix.os != 'macOS' && 'true' || 'false' }} # qt caches take too much space for macOS (1.1Gi)
 
     steps:
       - name: Checkout
         uses: actions/checkout@v5
         with:
           submodules: recursive
 
+      - name: Add msbuild to PATH
+        if: matrix.os == 'Windows'
+        id: add-msbuild
+        uses: microsoft/setup-msbuild@v2
+        with:
+          msbuild-architecture: x64
+
       # Using jianmingyong/ccache-action to setup ccache without using brew
       # It tries to download a binary of ccache from GitHub Release and falls back to building from source if it fails
       - name: Setup ccache
+        if: matrix.os == 'macOS'
         uses: jianmingyong/ccache-action@v1
         with:
           install-type: "binary"
-          ccache-key-prefix: ccache-${{matrix.os}}-${{matrix.soc}}-${{matrix.type}}
+          ccache-key-prefix: ccache-${{ matrix.runner }}-${{ matrix.soc }}-${{ matrix.type }}
+          max-size: 500M
           gh-token: ${{ secrets.GITHUB_TOKEN }}
 
-      # Using jurplel/install-qt-action to install Qt without using brew
-      # qt build using vcpkg either just fails or takes too long to build
-      - name: Install Qt ${{env.QT_VERSION}}
+      - name: Install Qt ${{ env.QT_VERSION }}
         uses: jurplel/install-qt-action@v4
         with:
-          cache: false # qt caches take too much space for macOS (1.1Gi)
-          version: ${{env.QT_VERSION}}
-          arch: ${{env.QT_ARCH}}
-          modules: ${{env.QT_MODULES}}
+          version: ${{ env.QT_VERSION }}
+          arch: ${{ env.QT_ARCH }}
+          modules: ${{ env.QT_MODULES }}
+          cache: ${{ env.CACHE_QT }}
 
       - name: Setup vcpkg cache
         id: vcpkg-cache
         uses: TAServers/vcpkg-cache@v3
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build on Xcode ${{matrix.xcode}}
-        shell: bash
+      - name: Configure macOS
+        if: matrix.os == 'macOS'
+        run: |
+          echo "PACKAGE_SUFFIX=-macOS${{ matrix.target }}${{ matrix.soc == 'Intel' && '_Intel' || '' }}${{ matrix.type == 'Debug' && '_Debug' || '' }}" >> $GITHUB_ENV
```

Comment:
honestly this is a bit ugly, but I don't know how to make it better

I decided to have separate, os-specific steps to set these environment variables because their logic is a bit too long. defining them directly in the `build` step would look very ugly IMO.

Reply:
if that's easier for you, sure, I don't particularly mind receiving updates though, it's also possible to simply use a parallel testing branch on your fork

Reply:
prs update when you force push to the branch though? like there is no big deal to make a new pr if you feel like it but it's not required for any reason

Reply:
> prs update when you force push to the branch though?

today I learned that closed PRs do not update when force-pushed

Reply:
github didn't allow to reopen a closed pr of branch that had been force pushed after the pr was closed